tgui/packages/tgui/interfaces/Manufacturer/index.tsx

```diff
@@ -0,0 +1,379 @@
+                </Stack.Item>
+              </Stack>
+            </Stack.Item>
+          </Stack>
+        </Button>
+      </Stack.Item>
+
+      <Stack.Item ml={0.5}>
+        <Stack vertical p={0} m={0}>
+          <Stack.Item pb={0.5}>
+            <Button align="center" width={2} height={2.5} icon="gear" pt={0.9} />
+          </Stack.Item>
+          <Stack.Item p={0} m={0}>
+            <Tooltip content={<BlueprintTooltip blueprint={blueprintData} />}>
+              <Button align="center" width={2} height={2.5} icon="question" pt={0.9} />
+            </Tooltip>
+          </Stack.Item>
+        </Stack>
+      </Stack.Item>
+
+    </Stack>
+  );
+};
+
+const MaterialRow = (props, context) => {
+  let { act } = useBackend(context);
+  let { resource, resourceAmt } = props;
+
+  return (
+    <TableRow collapsing>
+      <TableCell collapsing>
+        <Button icon="eject" onClick={() => act("material_eject", { "material": resource })} />
+      </TableCell>
+      <TableCell>
+        <Button icon="add" onClick={() => act("material_swap", { "resource": resource })} />
+      </TableCell>
+      <TableCell header>
+        {resource}
+      </TableCell>
+      <TableCell>
+        {resourceAmt/10}
+      </TableCell>
+    </TableRow>
+  );
+};
+
+const LoadedMaterials = (props, context) => {
+  let { resources } = props;
+  let resourceRows = [];
+  for (let resource in resources) {
+    resourceRows.push(<MaterialRow resource={resource} resourceAmt={resources[resource]} />);
+  }
+  return (
+    <Table>
+      {resourceRows}
+    </Table>
+  );
+};
+
+const CategoryDropdown = (props) => {
+  const { category, blueprints } = props;
+  let buttons = [];
+  for (let i in blueprints) {
+    buttons.push(<BlueprintButton name={i} blueprintData={blueprints[i]} />);
+  }
+
+  return (
+    <Collapsible open title={category}>
+      {buttons}
+    </Collapsible>
+  );
+};
+
+const CardInfo = (props, context) => {
+  const { data, act } = useBackend<ManufacturerData>(context);
+  return (data.card_owner === null || data.card_balance === null) ? (
+    <Flex backgroundColor={backgroundPop} py={1}>
+      <Flex.Item grow>
+        No Card Inserted
+      </Flex.Item>
+      <Flex.Item>
+        <Button icon="add" onClick={() => act("card", { "scan": true })}>Insert Card</Button>
+      </Flex.Item>
+    </Flex>
+  ) : (
+    <Box backgroundColor={backgroundPop} py={1}>
+      Card Owner: {data.card_owner}<br />
+      Current Balance: {data.card_balance}⪽
+      <Button icon="subtract" onClick={() => act("card", { "remove": true })}>Remove Card</Button>
+    </Box>
+  );
+};
+
+const is_set = (bits, bit) => { return bits & (1 << bit); };
+
+export const CollapsibleWireMenu = (props, context) => {
+  const { act, data } = useBackend<ManufacturerData>(context);
+  let wireContent = [];
+  let i = 0;
+  wireContent.push(
+    <>
+      <Flex style={{ "align-items": "center" }}>
+        <Flex.Item style={{ "width": "45%" }}>
```

Comment:
Percentage here is a little iffy, gets weird if people adjust the interface width.